mitodl/open-discussions
# Stricter ts linting, in-line with ocw-studio

#### Pre-Flight checklist


#### What are the relevant tickets?
https://github.com/mitodl/open-discussions/issues/3646

#### What's this PR do?
Makes all the new typescript code use the same linting config as ocw-studio, based on `eslint-config-mitodl`. (Well, almost the saame. I added `plugin:testing-library/react`.)

#### How should this be manually tested?
1. Make sure to install new dependencies, `docker compose run --rm watch yarn install`
2. These commands should all work:
    ```
    docker compose run --rm watch yarn run lint-fix
    docker compose run --rm watch yarn run lint-check
    docker compose run --rm watch yarn run fmt-fix
    docker compose run --rm watch yarn run fmt-check
    ```
3. If you want...Try making an error in a react hook, like adding an unnecessary dependency to `useCallback`. You should get an error. (_Well... A warning, apparently. Maybe that should be an error. But this PR at least brings this repo in-line with ocw-studio._)
<img width="1050" alt="Screen Shot 2022-07-19 at 6 35 19 PM" src="https://user-images.githubusercontent.com/9010790/179987880-ed166b6c-743b-4429-b210-e5d707a78bd5.png">
